src/test/java/org/testng/AssertTest.java

```diff
@@ -381,6 +381,12 @@ public void testAssertSame() {
     Assert.assertSame(object2, object);
   }
 
+  @Test(description = "GITHUB-2490")
+  public void testAssertNotEqualsWithNullObject() {
+    Assert.assertNotEquals(null, new Object());
+    Assert.assertNotEquals(new Object(), null);
+  }
+

```

Comment:
Might be good to additionally add two tests with a class which does not implement `equals(Object)` properly and verify that `assertNotEquals` fails (because the tests you added would also pass without the changes to `assertNotEquals`):
```java
static class BrokenEquals {
  @Override
  public void equals(Object o) {
    return true; // broken implementation
  }
}

@Test ...
public void testAssertNotEqualsBrokenEqualsExpectedNull() {
  Assert.assertNotEquals(new BrokenEquals(), null);
}


@Test ...
public void testAssertNotEqualsBrokenEqualsActualNull() {
  Assert.assertNotEquals(null, new BrokenEquals());
}
```
(maybe there is also a way to put this all into one test method?)

Reply:
👍 for the additional tests